idaes/core/util/model_diagnostics.py

```diff
@@ -1041,10 +1219,680 @@ def report_numerical_issues(self, stream=stdout):
+            _log.debug(
+                "Solver did not return an optimal termination condition for "
+                "Candidates MILP. This probably indicates the system is full rank."
+            )
+
+    def _prepare_ids_milp(self):
+        """
+        Prepare MILP to compute the irreducible degenerate set
+
+        """
+        _log.info("Building MILP model to compute irreducible degenerate set.")
+
+        n_eq = self.jacobian.shape[0]
+        n_var = self.jacobian.shape[1]
+
+        # Create Pyomo model for irreducible degenerate set
+        m_dh = ConcreteModel()
+
+        # Create index for constraints
+        m_dh.C = Set(initialize=range(n_eq))
+        m_dh.V = Set(initialize=range(n_var))
+
+        # Specify minimum size for nu to be considered non-zero
+        M = self.config.M
+
+        # Add variables
+        m_dh.nu = Var(m_dh.C, bounds=(-M, M), initialize=1.0)
+        m_dh.y = Var(m_dh.C, domain=Binary)
+
+        # Constraint to enforce set is degenerate
+        if issparse(self.jacobian):
+            J = self.jacobian.tocsc()
+
+            def eq_degenerate(m_dh, v):
+                # Find the columns with non-zero entries
+                C = find(J[:, v])[0]
+                return sum(J[c, v] * m_dh.nu[c] for c in C) == 0
+
+        else:
+            J = self.jacobian
+
+            def eq_degenerate(m_dh, v):
+                return sum(J[c, v] * m_dh.nu[c] for c in m_dh.C) == 0
+
+        m_dh.degenerate = Constraint(m_dh.V, rule=eq_degenerate)
+
+        def eq_lower(m_dh, c):
+            return -M * m_dh.y[c] <= m_dh.nu[c]
+
+        m_dh.lower = Constraint(m_dh.C, rule=eq_lower)
+
+        def eq_upper(m_dh, c):
+            return m_dh.nu[c] <= M * m_dh.y[c]
+
+        m_dh.upper = Constraint(m_dh.C, rule=eq_upper)
+
+        m_dh.obj = Objective(expr=sum(m_dh.y[c] for c in m_dh.C))
+
+        self.ids_milp = m_dh
+
+    def _solve_ids_milp(self, cons: Constraint, tee: bool = False):
+        """Solve MILP to check if equation 'cons' is a significant component
+        in an irreducible degenerate set
+
+        Args:
+            cons: constraint to consider
+            tee: Boolean, print solver output (default = False)
+
+        Returns:
+            ids: dictionary containing the IDS
+
+        """
+        _log.info(f"Solving IDS MILP for constraint {cons.name}.")
+        eq_con_list = self.nlp.get_pyomo_equality_constraints()
+        cons_idx = eq_con_list.index(cons)
+
+        # Fix weight on candidate equation
+        self.ids_milp.nu[cons_idx].fix(1.0)
+
+        # Solve MILP
+        results = self._get_solver().solve(self.ids_milp, tee=tee)
+
+        self.ids_milp.nu[cons_idx].unfix()
+
+        # Create empty dictionary
+        ids_ = {}
+
+        if check_optimal_termination(results):
+            # We found an irreducible degenerate set
+            for i in self.ids_milp.C:
+                # Check if constraint is included
+                if self.ids_milp.y[i]() > 0.9:
```

Comment:
As a general rule, it would be better to make the embedded constant (0.9) a named attribute (at the module scope?)